stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -645,7 +645,9 @@ AsyncTask *TaskGroupBase::claimWaitingTask() {
          "task is present!");
 
   auto waitingTask = waitQueue.load(std::memory_order_acquire);
-  if (!waitQueue.compare_exchange_strong(waitingTask, nullptr)) {
+  if (!waitQueue.compare_exchange_strong(waitingTask, nullptr,
```

Comment:
Why do we need a store release here? What other memory write are we trying to make visible before we publish this null ptr? I don't see any other memory write here?

Reply:
It is seq_cst by default so are you saying we can make this relaxed?

Reply:
> Why do we need a store release here? What other memory write are we trying to make visible before we publish this null ptr? I don't see any other memory write here?

The loads of other threads need to see the write to nullptr.